src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.cs

```diff
@@ -46,36 +46,34 @@ private abstract partial class CSharpCodeGenerator : CodeGenerator<StatementSynt
                 bool localFunction,
                 CancellationToken cancellationToken)
             {
-                var codeGenerator = Create(insertionPoint, selectionResult, analyzerResult, options, localFunction);
-                return codeGenerator.GenerateAsync(cancellationToken);
+                var codeGenerator = Create(selectionResult, analyzerResult, options, localFunction);
+                return codeGenerator.GenerateAsync(insertionPoint, cancellationToken);
```

Comment:
insertionpoint is expensive (it forks the document).  so it is now passed in as an arg, and only in the "actually create the new method" codepath.